tools/depcheck/README.md

```diff
@@ -0,0 +1,176 @@
+depcheck
+========
+
+Dependency analysis tool for Go repositories.
+
+### Why
+
+In the case of OpenShift, having a flat `vendor` dependency tree that brings in a
+large enough dependency, such as `kubernetes`, creates a problem with conflicting
+dependencies. It may be the case that we depend on a specific version of package `A`,
+while `kubernetes` (or one of its dependencies) depends on that same package `A`, 
+but a different version of it.
+Because only one version of package `A` may exist under the `vendor` directory, it
+might be useful to know:
+  - Which parts of our codebase depend on package `A`, and are responsible for the conflict?
+  - Would those parts of our codebase work with the version of package `A` that is brought
+  in by `kubernetes` (or one of its dependencies)?
+  - If not, can we break our dependency on package `A`?
+ 
+Both the `analyze` and `trace` commands included in this tool are meant to aid in 
+answering the above questions.
+
+### Getting started
+
+#### Limitations
+
+Before beginning to use this tool, it is important to note that because both the `trace`
+and `analyze` commands make use of [go list](https://golang.org/cmd/go/#hdr-List_packages)
+in order to obtain information about packages in a Go repository, there are limitations
+that exist for these two commands:  
+
+  - No symlinked directories will be traversed in a given repo. Any resulting graph information generated
+by these commands will simply not include these directories or their subtrees.
+
+  - A `GOPATH` must be set prior to their use. This allows each command to deal with paths
+  non-relative to the current working directory. It also allows for a dependency analysis
+  of a repository to occur in its own custom environment.
+
+As a result, of these limitations, it is recommended that any symlinks in a repo be "expanded" before using these
+two commands.
+
+If you are dealing with a repository that is free of symlinks, feel free to skip to the
+next section `Compiling`.
+
+A straightforward way of achieving this is to use `cp -L` to copy the original repository
+to a temporary location, while replacing symbolic links with the data that each one points to:
+
+```bash
+# setup the new temporary location
+$ mkdir -p /tmp/depcheck/src/github.com/openshift
+$ cp -rL /path/to/openshift/origin /tmp/depcheck/src/github.com/openshift/origin
+
+# set the GOPATH to look in the new temporary location, in order
+# to ensure we traverse packages of this particular directory, and
+# not the original "origin" directory
+$ export GOPATH=/tmp/depcheck
+```
+
+#### Compiling
+
+Finally, compile the `depcheck` binary:
+
+```bash
+$ go build /path/to/openshift/origin/tools/depcheck/depcheck.go
+```
+
+After these steps, you should have a `depcheck` binary, and a Go environment suitable for
+running the `trace` and `analyze` commands. 
+
+### Filters
```

Comment:
I built it, but how do I run it? A running section is needed, and only after that explain filters and excludes. I got lost jumping from compilation straight into filters.

Reply:
Thanks, will add a `Running` section